[WIP] associate colors to environments

MathieuDreano opened the pull request:
# Description

https://github.com/usebruno/bruno/issues/304
Associate a color to each environment, for quick visualization (and avoid calling your production environment by mistake...)
![bruno-env-color](https://github.com/usebruno/bruno/assets/12009156/3ca99ad0-b1fa-434e-b433-c8d23ed685cc)

I will not have more time to work on it, this is to share what's already been done.
TODO:
- color not rerendered in the droplist (only when restarting bruno)
- no test...

### Contribution Checklist:

- [x ] **The pull request only addresses one issue or adds one feature.**
- [ ] **The pull request does not introduce any breaking changes** --> well, I hope so
- [x ] **I have added screenshots or gifs to help explain the change if applicable.**
- [x ] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [x ] **Create an issue and link to the pull request.**

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

### Publishing to New Package Managers

Please see [here](../publishing.md) for more information.

MathieuDreano commented:
converted to draft, introduced bug, newly created env not visible on EnvironmentList

aphexcx commented:
Hi! When will this be merged? This one feature is the only thing stopping us from moving from Insomnia to Bruno LOL

MathieuDreano commented:
quite some changes, using redux store.
I think it's ready, feel free to try it out. I personally use it in my daily job.

MathieuDreano commented:
As requested, added visual aid on request tab
![image](https://github.com/usebruno/bruno/assets/12009156/6309a48b-ab3e-4cdb-bc34-78c13c3b01ff)

aphexcx commented:
Following up on this, really appreciate seeing the progress:)

aphexcx commented:
Checking in again:)

aphexcx commented:
Friendly ping:)

bobzomer commented:
> Friendly ping:)

I would also love it to see it merged and available in my beloved bruno!

aphexcx commented:
Beep 🤖

florianlacreuse commented:
Any updates on this PR? It would be nice to have it in Bruno. Thanks!

aeris commented:
If something interested, I rebase this PR on v1.23.0
https://git.imirhil.fr/aeris/bruno/src/branch/feature/environment-color

MathieuDreano commented:
Hello there, I rebased the work onto main branch. (not something I will do every month until the PR is closed)

peterkeppert-anasoft commented:
Hello, could somebody from Bruno maintainers please indicate what is the expected timeframe of this feature's integration or what are the next steps with this pull request?

Certainly for our team (and I believe many others) used to Insomnia's environment colors, this represents a significant and eagerly awaited UX improvement. Thanks!

SZubarev commented:
Any updates on this feature?

MathieuDreano commented on package.json:
```diff
@@ -25,6 +25,7 @@
     "jest": "^29.2.0",
     "pretty-quick": "^3.1.3",
     "randomstring": "^1.2.2",
+    "react-color": "^2.19.3",
```
react-color-circle is enough

MathieuDreano commented on packages/bruno-app/src/components/Environments/EnvironmentSettings/EnvironmentList/EnvironmentDetails/EnvironmentColor/index.js:
```diff
@@ -0,0 +1,67 @@
+const EnvironmentColor = ({ environment, collectionUid }) => {
+  const dispatch = useDispatch();
+
+  const formik = useFormik({
```
No need for formik here, can be simplified

MathieuDreano commented on packages/bruno-app/src/components/Environments/EnvironmentSettings/index.js:
```diff
@@ -43,11 +43,12 @@ const DefaultTab = ({ setTab }) => {
+  const [isModified, setIsModified] = useState(false);
   const [tab, setTab] = useState('default');
-  if (!environments || !environments.length) {
+  const [openCreateModal, setOpenCreateModal] = useState(false);
```
Remove these two lines, bad call inflict resolution

MathieuDreano commented:
Hello there, I rebased the branch on master andfixed the conflicts

Also, I changed to a border instead of full background (to match both dark and light mode)
And some codefix for maintainability.
I feel like the color could be used elsewhere to emphasis the environment, but let's keep this feature simple for now.

I still hope this will get released someday (I see the number of PR decreasing, that's a good trend)

helloanoop commented:
We’re onboard to add support for custom environment colors in Bruno. However, this effort will be revisited after completing YAML support for storing requests (https://github.com/usebruno/bruno/issues/3577), as it is essential for incorporating color information into the environment.

MatthiasVanMeerbeek commented:
> I feel like the color could be used elsewhere to emphasis the environment, but let's keep this feature simple for now.
I was thinking the same. Where you are looking when sending a request is of course the "send" (arrow ➡️) button. So maybe an idea would be to add a border with the color around the send button?

aeris commented:
Hello! Any ETA for this MR?
Env color is pretty usefull to quickly spot production area, and this MR seems not easy to port to new version, so need to stick to old 1.x version to keep colors.

MathieuDreano commented:
Hello it turns out i do have some time today to solve conflicts, working on it.
I'd be proud to be the oldest PR, and this is not a feature blocking from using any power of bruno, more like a nice to have.
Good job on bruno team to keep the tool alive.  ❤️

dan-lind commented:
> YAML support is being worked on. We should have something out late this month. Allocating bandwidth to review and get this merged in Nov.
> 
> cc @anusree-bruno

Having just fat-fingered a request to live instead of dev, I'm wondering what the state is here? November is almost over.

mdreano commented on packages/bruno-app/src/components/Environments/EnvironmentSettings/EnvironmentList/index.js:
```diff
@@ -12,15 +12,16 @@ import ToolHint from 'components/ToolHint';
 const EnvironmentList = ({ selectedEnvironment, setSelectedEnvironment, collection, isModified, setIsModified, onClose, setShowExportModal }) => {
-  const { environments } = collection;
+  const { environments, activeEnvironmentUid } = collection;
+  const [selectedEnvironment, setSelectedEnvironment] = useState(null);
```
done

mdreano commented on packages/bruno-app/package.json:
```diff
@@ -64,6 +64,7 @@
     "qs": "^6.11.0",
     "query-string": "^7.0.1",
     "react": "19.0.0",
+    "react-color": "^2.19.3",
```
done

sid-bruno commented:
Hey, @MathieuDreano thanks for the contribution, there's internal work going on to improve and polish your PR, would you mind if the contributors adopted this PR and merged it in?

MathieuDreano commented:
You're welcome, I looked at your "polish", you did like 90% of the job 😅
Congratulations to you.
